### DISCUSSION TITLE
How to create validation error for parent AST node?

### DISCUSSION THREAD

**Discussion by durianwaffle** [eclipse-langium/langium#1703] [Q&A] [answered]
Hi Langium community,

I can only create validation error for node's children property not its parents.
Is there any way to create validation error for a node's parent property?
```typescript
if (leftType !== rightType) {
  accept('error', `Can’t assign ${rightType} type to variable with type ${leftType} in assignment.`, 
  { node: assignemntStatement, property: 'left' });
}
```
For example, if the **operands have different types** in a**ssignment statement**, show **error line for the entire line not just on `left/op/right`**

And I am also wondering **how to debug for validator since there is no output**.

**Comment by ydaveluy** [comment, marked as answer]
Hello, if you remove the property attribute the entire statement will be on error:
```ts
if (leftType !== rightType) {
  accept('error', `Can’t assign ${rightType} type to variable with type ${leftType} in assignment.`, 
  { node: assignemntStatement });
}
```

**Comment by JohannesMeierSE** [comment]
Hello @durianwaffle,

you can use nearly any `AstNode` for the `node` property. Depending on your DSL, that could be e.g. `assignemntStatement.value` or `assignemntStatement.$container` or what ever you like. The `property` depends on the chosen value for `node`.
